### DISCUSSION TITLE
About project's creation

### DISCUSSION THREAD

**Discussion by mislavmatijevic** [foivz/pop-app-backend#2] [Show and tell]
@antoniokunstek I created the project using the following dependencies:

<img src="https://github.com/foivz/pop-app-backend/assets/11699688/a44cec5d-1bf8-46f4-a58c-61a856856953" width="1000">

_H2 Database_ - Quick testing of database access
_Spring Boot Devtools_ - This should enable fast redeployment (who doesn't like LiveReload?)
_Spring Web_ - Sets up Tomcat server for everything to work beautifully
_Spring Security_ - Hey, this one's on you 😃 
_MySQL Driver_ - Final DB is 99.99% going to be MySQL, so I figured better to put the driver in early on. Also, it could enable us to test the production enviroment early on, as well! I've got the server ready and waiting.
_Spring REST Docs_ - FIgured it'd be cool to have documentation powered by some Spring magic. Hopefully it's gonna pay off.

After creating the project, I only had to add line 126 (`<version>${kotlin.version}</version>`) to _pom.xml_ for IntelliJ to calm down and stop being a drama queen saying that the file is "not working" although everything launched correctly anyway.

Also, I had to Ctrl+Alt+Shift+S and set the correct JDK (JDK 17) as on a picture below.
![slika](https://github.com/foivz/pop-app-backend/assets/11699688/ffa00831-4b68-4678-8298-803a0134a8d5)

**Comment by mislavmatijevic** [comment]
The repository itself is created via command `git flow init` 😉

**Comment by antoniokunstek** [comment]
Dear Mislav,

I wanted to inform you that I have created a new feature branch named "add-h2-database."

I have initiated the development of the backend for this application. To begin with, I have created multiple application.properties files that dynamically modify variables based on the environment. For instance, during application development, it connects to an H2 database, while in production, it connects to a MySQL database. I have added a new dependency to facilitate the connection with the H2 database. Although spring-starter-jdbc already integrates this functionality and can automatically establish the connection, I have chosen an alternative approach by manually configuring the connection using DataSourceBuilder. Additionally, I have developed a simple controller to test the predefined structure of the response object (we still need to finalize its design). My next step is to create SQL scripts for generating and populating the database. Do you happen to have an ERA model that I can utilize as a basis for creating these scripts?

Furthermore, could you please provide me with the preferred commit structure you would like me to adhere to while working on this project?

I must admit that Kotlin is an exceptionally captivating programming language.

Thank you in advance for your response, and I hope the remainder of your day is pleasant.

Best regards,
Antonio Kunštek

**Reply by mislavmatijevic** [reply]
Dear Antonio,

your response made my otherwise mediocre day much more pleasant!

[ER diagram](https://github.com/foivz/pop-app-backend/wiki/ER-Diagram) is added in the Wiki page. Below the new V1 version of the initial ER diagram proposal, you can see my reasoning behind the two decisions I made. Feel free to discuss anything that catches your eye on the ER diagram in the new discussion (#4) dedicated to setting up the new database.

You can find [the preferred commit structure](https://github.com/foivz/pop-app-backend/wiki/Instructions-on-how-to-work-on-the-repo#committing) there, as well.

Ping me if you want the MySQL Workbench file of the ER diagram. It's standing by - I just haven't found the way to upload the file up here.

Looking forward to your comments! 😄

**Comment by mislavmatijevic** [comment]
Alright, so, I decided to be completely crazy and [record](https://youtu.be/-QIdN67dp6k) my implementation of issue #10, and the creation of PR #30. From start to finish.
Why is this crazy, you ask?
Because I just spent 3 hours non-stop programming and speaking about what I was doing. My throat hurts and I feel sleepy. But, well, at least my thought process is recorded and I find that somewhat cool. Also, I recorded how I handle various challenges during the implementation of this project, which might be useful for anyone who comes here after I'm gone.
